### PR TITLE
Scan for package descriptors once

### PR DESCRIPTION
**Pull request by yahavi**
- [x] All [tests](https://github.com/jfrog/jfrog-vscode-extension#building-and-testing-the-sources) passed. If this feature is not already covered by the tests, I added new tests.
- [x] I used `npm run format` for formatting the code before submitting the pull request.
-----

Instead of running `vscode.workspace.findFiles` 5 times for each project type, run it once before all scanners.
This change is expected to improve performance in big workspaces.

### CODE REVIEW

File: src/main/treeDataProviders/dependenciesTree/dependenciesTreeFactory.ts [hunk 1]
```diff
@@ -18,11 +19,41 @@ export class DependenciesTreesFactory {
...
+                quickScan
+            );
+            await NpmUtils.createDependenciesTrees(
+                projectDescriptors.get(PackageDescriptorType.NPM),
+                componentsToScan,
+                treesManager,
+                parent,
+                quickScan
+            );
+            if (!!projectDescriptors.get(PackageDescriptorType.PYTHON)) {
```
Comment (eyalbe4) [inline src/main/treeDataProviders/dependenciesTree/dependenciesTreeFactory.ts:40]: Any reason why we have this if statement for python only?

File: src/main/treeDataProviders/dependenciesTree/dependenciesTreeFactory.ts [hunk 2]
```diff
@@ -18,11 +19,41 @@ export class DependenciesTreesFactory {
...
+                await PypiUtils.createDependenciesTrees(workspaceFolders, componentsToScan, treesManager, parent, quickScan);
```
Comment (eyalbe4) [inline src/main/treeDataProviders/dependenciesTree/dependenciesTreeFactory.ts:41]: This line's arguments alignment is horizontal, while this call for the other package manages (see above) is vertical.

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

File: src/main/treeDataProviders/dependenciesTree/dependenciesTreeFactory.ts (hunk 1 above)
Comment (yahavi) [reply src/main/treeDataProviders/dependenciesTree/dependenciesTreeFactory.ts]: I moved this check to be in the `createDependenciesTrees`, to be consistent with the others.

File: src/main/treeDataProviders/dependenciesTree/dependenciesTreeFactory.ts (hunk 2 above)
Comment (yahavi) [reply src/main/treeDataProviders/dependenciesTree/dependenciesTreeFactory.ts]: It is a decision of the Prettier

**Pull request merged**